apis/python/node/src/lib.rs

```diff
@@ -1,48 +1,75 @@
 #![allow(clippy::borrow_deref_ref)] // clippy warns about code generated by #[pymethods]
 
-use dora_node_api::{dora_core::config::NodeId, DoraNode, Input};
+use dora_node_api::{
+    daemon::{Event, EventStream},
+    DoraNode,
+};
 use dora_operator_api_python::{metadata_to_pydict, pydict_to_metadata};
 use eyre::{Context, Result};
-use flume::Receiver;
 use pyo3::{
     prelude::*,
     types::{PyBytes, PyDict},
 };
 
 #[pyclass]
 pub struct Node {
-    id: NodeId,
-    inputs: Receiver<Input>,
+    events: EventStream,
     node: DoraNode,
 }
 
-pub struct PyInput(Input);
+pub struct PyInput<'a>(Event<'a>);
 
-impl IntoPy<PyObject> for PyInput {
+impl IntoPy<PyObject> for PyInput<'_> {
     fn into_py(self, py: Python) -> PyObject {
-        (
-            self.0.id.to_string(),
-            PyBytes::new(py, &self.0.data()),
-            metadata_to_pydict(self.0.metadata(), py),
-        )
-            .into_py(py)
+        let dict = PyDict::new(py);
+
+        let ty = match self.0 {
+            Event::Stop => "stop",
```

Comment:
Good point! The `stop` event is sent when a dataflow is stopped early through the CLI. I thought that some nodes might want to react to that in some way, e.g. to do some cleanup. We also send an `input-closed` event for every input that is closed in case the user only wants to wait on a subset of inputs.

However, I agree that we shouldn't force users to handle these events. I pushed https://github.com/dora-rs/dora/pull/162/commits/5666895a7c5d6445f6577795466e9506fef52fba to make these two event types optional. The event channel will be automatically closed by the daemon once all inputs of a node were closed, or when a `stop` command is received.